### PR TITLE
Adopt the int module

### PR DESCRIPTION
**Pull request by dorafmon**
Some questions I have along the way:

1. Should we keep formatting functions like `pp`? It's probably needed later.
2. Should we keep the Random number generators?

### CODE REVIEW

File: src/types/int.ml [hunk 1]
```diff
@@ -29,36 +29,17 @@ let pow a b =
...
+let fromString s =
   try Some (int_of_string s)
   with _ -> None
 
+let toString = string_of_int
+
 type output = char -> unit
 
 (* abstract printer *)
 let to_binary_gen (out:output) n =
```
Comment (glennsl) [inline src/types/int.ml]: might as well camelCase this too

File: src/types/int.mli [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  val (<) : t -> t -> bool
+
+  val (>) : t -> t -> bool
+
+  val (<=) : t -> t -> bool
+
+  val (>=) : t -> t -> bool
+end
+
+include module type of Infix
```
Comment (glennsl) [inline src/types/int.mli]: Won't this cause warnings if the operators are used?

File: tests/int_test.ml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    describe "Int" (fun () -> [
+        test "pow" (fun () -> 
+            pow 2 1 |> Expect.toEqual 2
+        );
+        test "pow - raise" (fun () ->
+            (fun () -> pow 0 0) |> Expect.toRaise
+        );
+        test "pow - negative" (fun () -> 
+            (fun () -> pow 1 (~-1)) |> Expect.toRaise
+        );
```
Comment (glennsl) [inline tests/int_test.ml]: I like to group the tests for the same functions by adding a blank line to separate them from the tests of other functions.

File: tests/int_test.ml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        test "fromString - None" (fun () ->
+            fromString "23a" |> Expect.toEqual (None)
+        );
+        test "toBinaryString - 1" (fun () ->
+            toBinaryString 1 |> Expect.toEqual ("0b1")
+        );
+        test "toBinaryString - 32" (fun () ->
+            toBinaryString 32 |> Expect.toEqual ("0b100000")
+        )
+    ])
```
Comment (glennsl) [inline tests/int_test.ml]: This whole file seems to be using 4-space indent. Or maybe ocp-indent is screwing up. Either way, we should stick with 2-space indents.

**Comment by glennsl** [comment]
I don't think either the random function or format functions are needed in each module. The Format module can probably be dropped entirely for now, and it seems more intuitive to me for the Random module to include int/char etc. generator functions

File: tests/int_test.ml (hunk 4 above)
Comment (glennsl) [inline tests/int_test.ml]: And tests for all the functions would be nice

File: src/types/int.mli (hunk 2 above)
Comment (dorafmon) [inline src/types/int.mli]: Well you have to use `open! Int.Infix`, I do not really see much value of these operators, I will just delete them.

Comment (glennsl) [inline src/types/int.mli]: `open! Int.Infix` is fine, though I don't see what usefulness it has, but this line includes `Infix` in `Int`, so you'll get them if you `open Int` too, and would probably include the warnings too.

Comment (bordoley) [inline src/types/int.mli]: Can I make the suggestion of calling this submodule Operators instead of Infix?

File: src/core/show.ml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 
 module type S = sig
```
Comment (bordoley) [inline src/core/show.ml]: Can we rename the module Stringifiable?

Comment (dorafmon) [inline src/core/show.ml]: Done, I am thinking to add a new module as well to complete the `fromString/toString` cycle, do you think we should call it `Parsable`? Or any other name you prefer?

**Comment by dorafmon** [comment]
Thanks!

**Pull request merged**